lib/src/common/github.dart

```diff
@@ -29,12 +30,24 @@ class GitHub {
   static const _ratelimitResetHeader = 'x-ratelimit-reset';
   static const _ratelimitRemainingHeader = 'x-ratelimit-remaining';
 
+  @visibleForTesting
+  static const versionHeader = 'X-GitHub-Api-Version';
+
   /// Authentication Information
   Authentication? auth;
 
   /// API Endpoint
   final String endpoint;
 
+  /// Calendar version of the GitHub API to use.
+  ///
+  /// Changing this value is unsupported. However, it may unblock you if there's
+  /// hotfix versions.
+  ///
+  /// See also:
+  ///   * https://docs.github.com/en/rest/overview/api-versions?apiVersion=2022-11-28
+  final String version;
```

Comment:
Most of this change was me debating if we should make this private. I lean towards making it public to give an easy escape hatch for downstream dependencies to override what the library is using.